### PR TITLE
Select component

### PR DESCRIPTION
**Pull request by saurabhsutar192**
Select Component

## Description
Select component is a combobox that supports selecting or searching and selecting and many more things

## Motivation and Context
A versatile select box that support many inbuilt functionalities

## How Has This Been Tested?
Manual and Local Build successful

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lib/src/components/ComboBox/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+} from "./select.css";
+import { SelectPropsType, OptionsType } from "./select.types";
+import { ArrowDown, Clear } from "./components/SelectIcons";
+
+const SelectComponent: ForwardRefRenderFunction<
+  HTMLDivElement,
+  SelectPropsType
+> = (
+  {
+    placeholder = "Pick one",
```
Comment (pruthvi2103) [inline lib/src/components/ComboBox/Select.tsx:37]: Do not keep default placeholders, it results un-intentional behaviour keep it blank

File: lib/src/components/ComboBox/Select.tsx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    options,
+    value,
+    height = "40px",
+    width = "100%",
+    roundness = "0",
```
Comment (pruthvi2103) [inline lib/src/components/ComboBox/Select.tsx:42]: isnt borderRadius a better name?

File: lib/src/components/ComboBox/Select.tsx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    color = "#2F80ED",
+    maxDropDownHeight = "auto",
+    onChange = () => {},
+    isSearchable = false,
+    isClearable = false,
+    isDisabled = false,
+    isMulti = false,
+    DropIcon,
+    error = false,
+    nothingFoundLabel = "Nothing Found!",
```
Comment (pruthvi2103) [inline lib/src/components/ComboBox/Select.tsx:52]: again do not add default text here

File: lib/src/components/ComboBox/Select.tsx [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+    onDropDownClose = () => {},
+    onDropDownOpen = () => {},
```
Comment (pruthvi2103) [inline lib/src/components/ComboBox/Select.tsx:54]: nitpick: do not add default value here, if its not provided let it be undefined

File: lib/src/index.ts [hunk 5]
```diff
@@ -15,3 +15,4 @@ export * from "./components/TextArea";
 export * from "./components/Tab";
 export * from "./components/Avatar";
 export * from "./components/Table";
+export * from "./components/ComboBox";
```
Comment (pruthvi2103) [inline lib/src/index.ts:18]: we decided to change name to Select right?

File: lib/src/hooks/useClickOutside.tsx [hunk 6]
```diff
@@ -27,7 +28,7 @@ function useClickOutside<T extends HTMLElement = HTMLElement>(
     };
 
     // Reload only if ref or handler changes
-  }, [ref, handler]);
+  }, [ref, handler, isCallbackEnabled]);
```
Comment (pruthvi2103) [inline lib/src/hooks/useClickOutside.tsx:31]: nice one!

**Review by pruthvi2103 — CHANGES_REQUESTED** [review]
Some comments, also while testing i found following problems:

During tab navigation, pressing space does not open the select.

File: lib/src/components/ComboBox/Select.tsx (hunk 1 above)
Comment (saurabhsutar192) [reply lib/src/components/ComboBox/Select.tsx:37]: ok

File: lib/src/index.ts (hunk 5 above)
Comment (saurabhsutar192) [reply lib/src/index.ts:18]: forgot to rename the folder

File: lib/src/icons/Icons.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      fill="none"
+      strokeLinecap="round"
+      strokeLinejoin="round"
+    >
+      <path stroke="none" d="M0 0h24v24H0z" fill="none"></path>
+      <circle cx={12} cy={12} r={9}></circle>
+      <path d="M10 10l4 4m0 -4l-4 4"></path>
+    </svg>
+  );
+};
```
Comment (pruthvi2103) [inline lib/src/icons/Icons.tsx:41]: shift this to `lib/src/components/_internal/Icons`

**Review by pruthvi2103 — CHANGES_REQUESTED** [review]
one change

**Review by pruthvi2103 — APPROVED** [review]
LGTM

**Pull request merged**